ts/components/BackendProgressIndicator.svelte

```diff
@@ -0,0 +1,115 @@
+<!--
+Copyright: Ankitects Pty Ltd and contributors
+License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+-->
+<script lang="ts">
+    import type { Progress } from "@tslib/anki/collection_pb";
+    import {
+        OpChanges,
+        type OpChangesAfterUndo,
+        type OpChangesWithCount,
+        type OpChangesWithId,
+    } from "@tslib/anki/collection_pb";
+    import type { ImportResponse } from "@tslib/anki/import_export_pb";
+    import { bridgeCommand } from "@tslib/bridgecommand";
+    import { runWithBackendProgress } from "@tslib/progress";
+
+    import { pageTheme } from "../sveltelib/theme";
+
+    type ResultWithChanges =
+        | OpChanges
+        | OpChangesWithCount
+        | OpChangesWithId
+        | OpChangesAfterUndo
+        | ImportResponse
+        | { changes: OpChanges };
```

Comment:
Listing out all the cases is fine too, but we could probably get away with just

```
    type ResultWithChanges =
        | OpChanges
        | { changes: OpChanges };
```

The reason they're all listed out in the Python code is to work around mypy/protobuf limitations.